receiver/otlpreceiver/otlp.go

```diff
@@ -230,6 +236,22 @@ func (r *otlpReceiver) registerLogsConsumer(lc consumer.Logs) error {
 		r.httpMux.HandleFunc("/v1/logs", func(w http.ResponseWriter, req *http.Request) {
 			handleLogs(w, req, r.logReceiver, jsEncoder)
 		}).Methods(http.MethodPost).Headers("Content-Type", jsonContentType)
+		r.httpMux.HandleFunc("/v1/logs", func(resp http.ResponseWriter, req *http.Request) {
+			handleUnmatchedRequests(resp, req)
+		})
 	}
 	return nil
 }
+
+func handleUnmatchedRequests(resp http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodPatch {
+		status := http.StatusMethodNotAllowed
+		writeResponse(resp, "text/plain", status, []byte(fmt.Sprintf("%v method not allowed, supported: [GET, POST]", status)))
```

Comment:
Are GET requests supported? I'm not seeing where that's handled.

Reply:
You're right. The issue originally talked about `PATCH`, but I can expend it to everything but `POST`.

Reply:
@codeboten This is ready for a re-review when you're ready.